wallet/src/Pos/Util/BackupPhrase.hs

```diff
@@ -29,20 +30,17 @@ newtype BackupPhrase = BackupPhrase
     } deriving (Eq, Generic)
 
 instance Arbitrary BackupPhrase where
-    arbitrary = BackupPhrase <$> vectorOf 12 (elements englishWords)
+    arbitrary = do
+        em <- arbitraryMnemonic 16
+        case em of
+            Left _  -> arbitrary
+            Right a -> pure a
     shrink    = genericShrink
 
--- | (Some) valid English words as taken from <https://github.com/bitcoin/bips/blob/master/bip-0039.mediawiki BIP-39>
-englishWords :: [Text]
-englishWords = [ "recycle" , "child" , "universe" , "extend" , "edge" , "tourist"
-               , "swamp" , "rare" , "enhance" , "rabbit" , "blast" , "plastic" , "attitude"
-               , "name" , "skull" , "merit" , "night" , "idle" , "bone" , "exact"
-               , "inflict" , "legal" , "predict" , "certain" , "napkin" , "blood"
-               , "color" , "screen" , "birth" , "detect" , "summer" , "palm"
-               , "entry" , "swing" , "fit" , "garden" , "trick" , "timber"
-               , "toss" , "atom" , "kitten" , "flush" , "master" , "transfer"
-               , "success" , "worry" , "rural" , "silver" , "invest" , "mean"
-               ]
+arbitraryMnemonic :: Int -> Gen (Either Text BackupPhrase)
```

Comment:
Agreed. 

> Now, I understand that my implementation of octets is naive because I think the size of the checksum is proportional to the number of words chosen

for 9 words mnemonic we would need 12 octets - but yes - it is deterministic.

Lets track this under https://iohk.myjetbrains.com/youtrack/issue/CSL-2441